sensors/cron_sensor.py

```diff
@@ -0,0 +1,291 @@
+                result = self.check_enforcements(enforcements, previous_cron_dt, next_cron_dt)
+
+                if result:
+                    self.delete_from_kv(rule.ref)
+            else:
+                st2_comments = "Cron job is not running and no enforcements can be found"
+                self.dispatch_trigger(st2_rule_name=rule.ref,
+                                      st2_server=self.st2_fqdn,
+                                      st2_comments=st2_comments)
+
+        self._sensor_service.set_value(name=self.kv_sensor_name, value=self.kv_enforcements)
+
+    def check_enforcements(self, enforcements, previous_cron, next_cron):
+        """ Checks all the enforments to find if the cron was executed and to
+        get the status of the execution if one can be found
+        """
+        for enforcement in enforcements:
+            # Parse the enforcement enforced time to be in datetime format for comparison
+            # We need to drop the microseconds from the datetime format for the comparison
+            # to be accurate
+            enforced_at = parse(enforcement.enforced_at).replace(microsecond=0)
+
+            # Creating a buffer of one run time
+            cron_delta = next_cron - previous_cron
+            cron_low_buffer = previous_cron - cron_delta
+
+            if cron_low_buffer <= enforced_at <= next_cron:
+                # If the rule logic fails then no execution_id exists in the rule enforcement
+                if hasattr(enforcement, 'execution_id'):
+                    st2_execution = self.st2_client.liveactions.get_by_id(enforcement.execution_id)
+                    if st2_execution.status in PROGRESS_STATUSES:
+                        self._logger.info("Currently running execution. Will check next run")
+                        return False
+                    elif st2_execution.status in ERRORED_STATUSES:
+                        self.dispatch_trigger(st2_rule_name=enforcement.rule['ref'],
+                                              st2_server=self.st2_fqdn,
+                                              st2_enforcement_exe_id=enforcement.execution_id,
+                                              st2_comments="Cronjob execution failed",
+                                              st2_enforcement_id=enforcement.id)
+                        return False
+                    else:
+                        if enforcement.rule['ref'] in self.kv_enforcements:
+                            self.dispatch_trigger(st2_rule_name=enforcement.rule['ref'],
+                                                  st2_server=self.st2_fqdn,
+                                                  st2_enforcement_exe_id=enforcement.execution_id,
+                                                  st2_comments="Cronjob ran successfully",
+                                                  st2_enforcement_id=enforcement.id,
+                                                  st2_state="closed")
```

Comment:
Instead of `st2_state="closed"` let do the actual status. Like `st2_state="success"`